### PR TITLE
Use elements range generator

### PR DESCRIPTION
**Pull request by akva2**
Still prettier.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Review by blattms — APPROVED** [review]

File: opm/simulators/linalg/elementborderlistfromgrid.hh [hunk 1]
```diff
@@ -68,13 +69,9 @@ class ElementBorderListFromGrid
...
             , borderList_(borderList)
         {
-            auto elemIt = gridView_.template begin<0>();
-            const auto& elemEndIt = gridView_.template end<0>();
-            for (; elemIt != elemEndIt; ++elemIt) {
-                if (elemIt->partitionType() != Dune::InteriorEntity) {
-                    Index elemIdx = static_cast<Index>(map_.index(*elemIt));
-                    blackList_.addIndex(elemIdx);
-                }
+            for (const auto& elem : elements(gridView_, Dune::Partitions::interior)) {
```
Comment (akva2) [inline opm/simulators/linalg/elementborderlistfromgrid.hh:72]: i accidentially reversed the logic here. @blattms is there a Partitions entry to use for everything non-interior?

Comment (akva2) [reply opm/simulators/linalg/elementborderlistfromgrid.hh]: for now i restored the if.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Approved and build is green.  I'll merge into master.

**Pull request merged**